pkg/cache/cache.go

```diff
@@ -27,13 +26,16 @@ var (
 
 type localCache struct{}
 
-func (localCache) getDecision(clientIP string) (bool, error) {
+func (localCache) getDecision(clientIP string) (bool, bool, error) {
 	banned, isCached := cache.Get(clientIP)
 	bannedString, isValid := banned.(string)
+	logger.Debug(fmt.Sprintf("cache:getDecision:bannedString '%s', isValid '%t', isCached '%t'", bannedString, isValid, isCached))
 	if isCached && isValid && len(bannedString) > 0 {
-		return bannedString == cacheBannedValue, nil
+		isBanned := bannedString == cacheBannedValue || bannedString == cacheCaptchaValue
+		isCaptcha := bannedString == cacheCaptchaValue
```

Comment:
isBanned := bannedString != cacheNoBannedValue
isCaptcha := bannedString == cacheCaptchaValue

or  

isCaptcha := bannedString == cacheCaptchaValue
isBanned := bannedString == cacheBannedValue || isCaptcha

Reply:
Oh, didn't know Go can do that, thanks :)